content/blog/teach-tidyverse-2021/index.Rmd

```diff
@@ -0,0 +1,437 @@
+
+Another development that will help students, particularly those working on an assignment involving a local data file, create reprexes is the new `wd` argument to set working directory of the reprex.[^2]
+Writing a reproducible example with a minimal dataset is better practice, but this can be quite difficult for new learners.
+Being able to easily use local data will make it easier for them to benefit from other aspects of reprex earlier on.
+
+[^2]: Turns out this was always possible using the `outfile` argument, but now it's more intuitive!
+
+Being able to create a reprex in the current working directory means you can also benefit from a project-level `.Rprofile` if you happen to have one in your project.
+This is likely not going to have implications for new learners, for whom this would be an advanced concept, but it can be helpful for instructors who teach with a different suite of packages than what they locally have installed (e.g., CRAN versions for teaching vs. development versions for personal use).
+If this describes you, I recommend using [**renv**](https://rstudio.github.io/renv/index.html) in projects where you keep teaching materials, which uses `.Rprofile` to implement a project-specific package library.
+Then, `reprex(wd = ".")` can be used to render using the project-specific library.
+
+For more on updates in reprex, read the blog posts for the [1.0.0](https://www.tidyverse.org/blog/2021/02/reprex-1-0-0/) and [2.0.0](https://www.tidyverse.org/blog/2021/04/reprex-2-0-0/) releases.
+And if you're new to reprex, start [here](https://reprex.tidyverse.org/articles/articles/learn-reprex.html).
+
+## Building on tidyverse for modeling with tidymodels
+
+The **tidymodels** framework is a collection of packages for modeling and machine learning using tidyverse principles.
+This framework has been around since 2017, but over the past year many of the packages within tidymodels have become stable and gained lots of documentation, making them attractive choices for teaching.
+If you're introducing your students to data science with the tidyverse, a great next step to consider is using tidymodels when it comes to modeling and inference.
+
+```{r}
+library(tidymodels)
+```
+
+From a pedagogical perspective, tidymodels has three main advantages:
+
+1.  Similar interfaces to different models.
+2.  Model outputs as tibbles, which are straightforward to interact with for learners who already know how to wrangle and visualize data stored in this format.
+3.  Features that help users avoid common machine learning pitfalls.
```

Comment:
Can you make this more precise? i.e. avoiding over-fitting by making the test-training split a fundamental part of the workflow (except I wouldn't use workflow to avoid confusion with the workflows package).